src/Perlang.Interpreter/PerlangInterpreter.cs

```diff
@@ -1687,7 +1761,9 @@ public VoidObject VisitWhileStmt(Stmt.While stmt)
                 case STAR_STAR:
                     CheckNumberOperands(expr.Operator, left, right);
 
-                    if (left is float or double || right is float or double)
+                    if ((left is float or double && right is float or double) ||
+                        (left is float or double && right is int) ||
+                        (left is int && right is float or double))
                     {
                         double value = leftConvertible!.ToDouble(CultureInfo.InvariantCulture);
                         double exponent = rightConvertible!.ToDouble(CultureInfo.InvariantCulture);
```

Comment:
Unsure of why I did this change (I started working on it weeks ago but got distracted by #313 and related PRs), but I _think_ it was because `long` values cannot really be coerced to `double` like this without risk of precision loss. So that's why we wanted to narrow this down a bit, saying that `int`+`double` or `double`+`int` is fine, but `long`+`double` or vice versa is forbidden.

(In general, I think we should do a review of what type combinations we currently support with our binary operators. There are some of them being introduced now in this PR and some which were perhaps existing earlier already which might not be completely "safe" from an "avoid-precision-loss" perspective. I think I'll have to chew on that one for a while; it involves some very non-obvious tradeoffs.)